test/stage1/behavior/pointers.zig

```diff
@@ -288,3 +288,26 @@ test "pointer to array at fixed address" {
     // Silly check just to reference `array`
     expect(@ptrToInt(&array[0]) == 0x10);
 }
+
+test "pointer arithmetic affects the alignment" {
+    {
+        var ptr: [*]align(8) u32 = undefined;
+        var x: usize = 1;
+
+        expect(@typeInfo(@TypeOf(ptr)).Pointer.alignment == 8);
+        const ptr1 = ptr + 1; // 1 * 4 = 4 -> lcd(4,8) = 4
+        expect(@typeInfo(@TypeOf(ptr1)).Pointer.alignment == 4);
+        const ptr2 = ptr + 4; // 4 * 4 = 16 -> lcd(16,8) = 8
+        expect(@typeInfo(@TypeOf(ptr2)).Pointer.alignment == 8);
+        const ptr3 = ptr + 0; // no-op
+        expect(@typeInfo(@TypeOf(ptr3)).Pointer.alignment == 8);
+        const ptr4 = ptr + x; // runtime-known addend
+        expect(@typeInfo(@TypeOf(ptr4)).Pointer.alignment == 1);
```

Comment:
Wait, this should still be aligned to u32. `x` is runtime known but the number of bytes the pointer marches forward is `@sizeOf(u32) * x` which is guaranteed to be divisible by `@sizeOf(u32)`.

Reply:
Yeah, I focused too much on the comptime-known branch and forgot the other one, now it should be fine!